runtime/rxjava2-extensions/src/main/kotlin/com/squareup/sqldelight/runtime/rx/QueryObservable.kt

```diff
@@ -0,0 +1,71 @@
+package com.squareup.sqldelight.runtime.rx
+
+import com.squareup.sqldelight.Query
+import io.reactivex.Observable
+import io.reactivex.Observer
+import io.reactivex.Scheduler
+import io.reactivex.annotations.CheckReturnValue
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import java.util.Optional
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+ * Turns this [Query] into an [Observable] which emits whenever the underlying result set changes.
+ *
+ * ### Scheduler:
+ *   [observe] by default operates on the [io.reactivex.schedulers.Schedulers.io] scheduler but can
+ *   be optionally overridden with [scheduler]
+ */
+@CheckReturnValue
+fun <T : Any> Query<T>.observe(scheduler: Scheduler = Schedulers.io()): QueryObservable<T> {
+  return QueryObservable(this, scheduler)
+}
+
+class QueryObservable<RowType : Any>(
+  private val query: Query<RowType>,
+  private val scheduler: Scheduler
+) : Observable<Query<RowType>>() {
+  override fun subscribeActual(observer: Observer<in Query<RowType>>) {
+    val listener = Listener(query, observer, scheduler)
+    observer.onSubscribe(listener)
+    query.addListener(listener)
+    listener.queryResultsChanged()
+  }
+
+  @CheckReturnValue
+  fun mapToOne() = map { it.executeAsOne() }
+
+  @CheckReturnValue
+  fun mapToOneOrDefault(defaultValue: RowType) = map { it.executeAsOneOrNull() ?: defaultValue }
+
+  @CheckReturnValue
+  fun mapToOptional() = map { Optional.ofNullable(it.executeAsOneOrNull()) }
+
+  @CheckReturnValue
+  fun mapToList() = map { it.executeAsList() }
```

Comment:
Can you explicitly specify the return type of these 4 methods? Otherwise I believe the type becomes `Observable<List<RowType!>!>` instead of `Observable<List<RowType>>`.